src/theory/quantifiers/sygus/cegis_unif.cpp

```diff
@@ -83,16 +96,21 @@ bool CegisUnif::constructCandidates(const std::vector<Node>& enums,
                                     std::vector<Node>& candidate_values,
                                     std::vector<Node>& lems)
 {
-  Assert(enums.size() == enum_values.size());
-  if (enums.empty())
+  if (addEvalLemmas(enums, enum_values))
   {
+    Trace("cegis-unif-lemma") << "Added eval lemmas\n";
     return false;
   }
   unsigned min_term_size = 0;
   std::vector<unsigned> enum_consider;
+  NodeManager* nm = NodeManager::currentNM();
   Trace("cegis-unif-enum") << "Register new enumerated values :\n";
   for (unsigned i = 0, size = enums.size(); i < size; ++i)
   {
+    if (std::find(d_enums.begin(), d_enums.end(), enums[i]) == d_enums.end())
```

Comment:
Maybe document why we are doing this. It may also be helpful to rename d_enums -> d_cond_enums.